File: NOVACommerce.API/Models/Product.cs
namespace NOVACommerce.API.Models;

public class Product
{
    public int Id { get; set; }
    public String Name { get; set; } = string.Empty;
    public String? Description { get; set; }
    public decimal Price { get; set; }
    public String? ImageUrl { get; set; }
    public int StockQuantity { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public int CategoryId { get; set; }
    public Category Category { get; set; } = null!;
}

File: nova_commerce/lib/features/products/models/product.dart
class Product {
  final int id;
  final String name;
  final String? description;
  final double price;
  final String? imageUrl;
  final int stockQuantity;
  final int categoryId;
  final String categoryName;

  const Product({
    required this.id,
    required this.name,
    this.description,
    required this.price,
    this.imageUrl,
    required this.stockQuantity,
    required this.categoryId,
    required this.categoryName,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      price: (json['price'] as num).toDouble(),
      imageUrl: json['imageUrl'],
      stockQuantity: json['stockQuantity'],
      categoryId: json['categoryId'],
      categoryName: json['categoryName'],
    );
  }
}
